GVFS/GVFS.Common/Maintenance/PostFetchStep.cs

```diff
@@ -46,7 +46,12 @@ protected override void PerformMaintenance()
                 string commitGraphLockPath = Path.Combine(this.Context.Enlistment.GitObjectsRoot, "info", CommitGraphLock);
                 this.Context.FileSystem.TryDeleteFile(commitGraphLockPath);
 
-                this.RunGitCommand((process) => process.WriteCommitGraph(this.Context.Enlistment.GitObjectsRoot, this.packIndexes), nameof(GitProcess.WriteCommitGraph));
+                GitProcess.Result writeResult = this.RunGitCommand((process) => process.WriteCommitGraph(this.Context.Enlistment.GitObjectsRoot, this.packIndexes), nameof(GitProcess.WriteCommitGraph));
+
+                if (writeResult.ExitCodeIsFailure)
+                {
+                    this.LogErrorAndRewriteCommitGraph(activity, this.packIndexes);

```

Comment:
This logs error and attempts to write commit graph again, right? - wondering why it would succeed the second time.

Reply:
It deletes the commit-graph first.

Reply:
oh ok...